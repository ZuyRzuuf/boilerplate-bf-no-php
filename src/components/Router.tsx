import * as React from 'react';
import {
    BrowserRouter,
    Route
} from 'react-router-dom';
import {Switch} from 'react-router';

import AnotherPage from './AnotherPage';
import Dashboard from './Dashboard';
import Header from './Header';
import NotFound from './NotFound';

const Router = () => {
    return (
        <BrowserRouter>
            <>
                <Header />

                <Switch>
                    <Route component={Dashboard} exact path="/" />
                    <Route component={AnotherPage} path="/another-page" />
                    <Route component={NotFound} />
                </Switch>
            </>
        </BrowserRouter>
    )
};

export default Router;
