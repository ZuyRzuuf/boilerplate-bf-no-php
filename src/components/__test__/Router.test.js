import * as React from 'react';
import { MemoryRouter } from 'react-router';

import Router from '../App';

describe('<Router /> suite', function() {
    test('renders isolated component correctly', () => {
        const wrapper = shallow(
            <Router />
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('renders HTML tree correctly', () => {
        const wrapper = render(
            <MemoryRouter initialEntries={[ '/' ]} initialIndex={0}>
                <Router />
            </MemoryRouter>
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('renders components tree correctly', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={[ '/' ]} initialIndex={0}>
                <Router />
            </MemoryRouter>
        );
        expect(wrapper).toMatchSnapshot();
    });
});
