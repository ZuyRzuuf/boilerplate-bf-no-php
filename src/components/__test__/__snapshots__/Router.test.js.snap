// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Router /> suite renders HTML tree correctly 1`] = `
<div>
  <div>
    Backend/Frontend Node, React, Typescript boilerplate
  </div>
  <div>
    <a
      href="/"
    >
      Dashboard
    </a>
    <a
      href="/another-page"
    >
      Another Page
    </a>
    <a
      href="/wrong-uri"
    >
      Wrong URI
    </a>
  </div>
  <div>
    Dashboard
  </div>
</div>
`;

exports[`<Router /> suite renders components tree correctly 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      "/",
    ]
  }
  initialIndex={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "42dedh",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "42dedh",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <App>
      <Router>
        <Component>
          <div>
            <Header>
              <div>
                Backend/Frontend Node, React, Typescript boilerplate
              </div>
              <div>
                <Link
                  replace={false}
                  to="/"
                >
                  <a
                    href="/"
                    onClick={[Function]}
                  >
                    Dashboard
                  </a>
                </Link>
                <Link
                  replace={false}
                  to="/another-page"
                >
                  <a
                    href="/another-page"
                    onClick={[Function]}
                  >
                    Another Page
                  </a>
                </Link>
                <Link
                  replace={false}
                  to="/wrong-uri"
                >
                  <a
                    href="/wrong-uri"
                    onClick={[Function]}
                  >
                    Wrong URI
                  </a>
                </Link>
              </div>
            </Header>
            <Switch>
              <Route
                component={[Function]}
                computedMatch={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
                exact={true}
                location={
                  Object {
                    "hash": "",
                    "key": "42dedh",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  }
                }
                path="/"
              >
                <Dashboard
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "canGo": [Function],
                      "createHref": [Function],
                      "entries": Array [
                        Object {
                          "hash": "",
                          "key": "42dedh",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        },
                      ],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "index": 0,
                      "length": 1,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "key": "42dedh",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  location={
                    Object {
                      "hash": "",
                      "key": "42dedh",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    }
                  }
                  match={
                    Object {
                      "isExact": true,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                >
                  <div>
                    Dashboard
                  </div>
                </Dashboard>
              </Route>
            </Switch>
          </div>
        </Component>
      </Router>
    </App>
  </Router>
</MemoryRouter>
`;

exports[`<Router /> suite renders isolated component correctly 1`] = `
<Fragment>
  <Router />
</Fragment>
`;
